crates/update-holochain-versions/src/main.rs

```diff
@@ -0,0 +1,421 @@
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let s = s.to_string();
+        let split = s
+            .splitn::<_>(2, ':')
+            .map(|s| s.to_lowercase())
+            .map(|s| s.trim().to_owned())
+            .map(|s| s.replace('"', ""))
+            .collect::<Vec<_>>();
+        match (split.get(0), split.get(1)) {
+            (Some(key), Some(branch)) if key == "branch" => Ok(GitRev::Branch(branch.clone())),
+            (Some(key), Some(id)) if key == "commit" => Ok(GitRev::Manual(id.clone())),
+            (_, _) => bail!("invalid git-rev provided: {}", s),
+        }
+    }
+}
+
+impl<'a> GitRev {
+    pub(crate) fn toml_src_value(&'a self) -> ([&'a str; 2], &'a str) {
+        match &self {
+            GitRev::Branch(branch) => (["src", "branch"], branch),
+            GitRev::Manual(commit) => (["src", "manual"], commit),
+        }
+    }
+
+    pub(crate) fn is_manual(&'a self) -> bool {
+        matches!(self, GitRev::Manual(_))
+    }
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct HolochainVersion<'a> {
+    url: &'a str,
+    rev: &'a str,
+    sha256: &'a str,
+    cargo_lock: CargoLock<'a>,
+    bins: Vec<String>,
+
+    lair: LairVersion<'a>,
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct LairVersion<'a> {
+    url: &'a str,
+    rev: &'a str,
+    sha256: &'a str,
+    cargo_lock: CargoLock<'a>,
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct CargoLock<'a> {
+    _lock_file: &'a str,
+    output_hashes: HashMap<String, String>,
+}
+
+#[cfg(features = "disabled")]
+fn git_rev_helper(
+    h: &handlebars::Helper,
+    _: &handlebars::Handlebars,
+    _: &handlebars::Context,
+    rc: &mut handlebars::RenderContext,
+    out: &mut dyn handlebars::Output,
+) -> handlebars::HelperResult {
+    let param = h.param(0).unwrap();
+
+    let obj = param
+        .value()
+        .as_object()
+        .unwrap()
+        .iter()
+        .next()
+        .ok_or_else(|| handlebars::RenderError::new("invalid GitRev"))?;
+
+    let git_rev = GitRev::from_str(&format!("{}:{}", obj.0, obj.1,)).unwrap();
+
+    let (k, v) = match git_rev {
+        GitRev::Branch(branch) => ("src.branch", branch),
+        GitRev::Commit(commit) => ("src.manual", commit),
+    };
+
+    out.write(&format!(r#"{} = "{}""#, k, v))?;
+    Ok(())
+}
+
+static HOLOCHAIN_VERSION_TEMPLATE: &str = "holochain_version_template";
+static HOLOCHAIN_VERSIONS_TEMPLATE: &str = "holochain_versions_template";
+
+static HANDLEBARS: Lazy<handlebars::Handlebars> = Lazy::new(|| {
+    let mut handlebars = handlebars::Handlebars::new();
+    handlebars
+        .register_template_string(
+            HOLOCHAIN_VERSION_TEMPLATE,
+            r#"# This file was generated.
+# TODO: add comment at the top how to generate the file or how it was generated
```

Comment:
I'm assuming you forgot about this TODO

Reply:
no, it's a legitimate TODO :smile:

Reply:
At the very least can you add a link to some documentation before merging this? And is this TODO tracked somewhere where we will remember it? If it's not important enough to track, it sort of seems like the TODO should be deleted.

Reply:
> And is this TODO tracked somewhere where we will remember it?

yes, [here](https://www51.v1host.com/Holo/Task.mvc/Summary?oidToken=Task%3A36356).




> At the very least can you add a link to some documentation before merging this?

yes, i didn't plan on merging before resolving the TODO